Diplom_3

avonurumanna opened the pull request.

avonurumanna committed:
add classes for generating user data

avonurumanna committed:
add pojo classes

avonurumanna committed:
add settings

avonurumanna committed:
add waiting class

avonurumanna committed:
add yandex driver

avonurumanna committed:
add settings

avonurumanna committed:
add ui tests

avonurumanna committed:
add report

pavel-ulan commented on src/main/java/page_object/AccountProfilePage.java:
```diff
@@ -0,0 +1,46 @@
+
+    private final By logoLink = By.xpath("//*[@id=\"root\"]/div/header/nav/div/a");
+
+    private final By constructorLink = By.xpath("//*[@id=\"root\"]/div/header/nav/ul/li[1]/a");
```
**надо исправить**
не не не, так точно не пойдет. Не используй локаторы состоящие из пути. Это очень плохое решение. Если что-то поменяется в верстке, то этот путь сломается (или что еще хуже, начнет вести к другому элементу).

pavel-ulan replied:
ну и к прочему - по такому локатору совершенно непонятно к чему он ведет (т.е. надо лезть и проверять каждый раз)

pavel-ulan commented on src/test/java/ui_tests/RegistrationTest.java:
```diff
@@ -0,0 +1,61 @@
+        RegistrationPage registrationPage = new RegistrationPage(driver);
+        registrationPage.fillEmail("testemail456@gmail.com");
+        registrationPage.fillName("test");
+        registrationPage.fillPassword("12345");
```
**можно лучше**
прослеживается по все работе. Не используй хардкод в тестах. Как правило, тестовые данные поставляют провайдеры. Если не используешь из, то выноси в виде константы. Тогда если надо что-то поменять, нет необходимости лезть в код каждого теста и править.

pavel-ulan reviewed (changes requested):
Работа хорошая, но проблема с локаторами (и это довольно серьезная проблема). Подумай, что правильнее использовать для каждого из них (соответствие текста, класс, стиль или что-то еще), но точно не оставляй путь до элемента.

avonurumanna committed:
changes according to the reviewer comments:
names of tests were changes according to format ***Test
locators were changed
for test data were created new variables

pavel-ulan reviewed (approved):
Да, молодец, сейчас все выглядит значительно лучше =)
